### PR TITLE
Add no-await-in-loop rule

### PR DESCRIPTION
**Pull request by magurotuna**
This PR is to add [no-await-in-loop - Rules - ESLint](https://eslint.org/docs/rules/no-await-in-loop#top) rule.
Since this rule is not listed as `recommended`, I added it only to `get_all_rules`.

### CODE REVIEW

File: src/rules/no_await_in_loop.rs [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use crate::test_util::*;
+
+  #[test]
+  fn no_await_in_loop_valid() {
```
Comment (disizali) [inline src/rules/no_await_in_loop.rs:210]: can you add this source to valid tests, please?
```ts
async function myLoop() {
  for (let i = await something(); i < n; i++) {}
}
```
This is considered an error,
That shouldn't be the case.

Comment (disizali) [inline src/rules/no_await_in_loop.rs:210]: The same goes for this source :
```ts
async function myLoop() {
  for (const thing of await anotherThing) {
  // also :
  // for (let thing in await anotherThing) {}
    await something();
  }
}
```

Comment (magurotuna) [reply src/rules/no_await_in_loop.rs:210]: Thank you for the review!
Oh, I missed that. I'll fix it.

Comment (magurotuna) [reply src/rules/no_await_in_loop.rs:210]: Thanks again. I'll fix as well.

**Comment by magurotuna** [comment]
I've added more valid testcases:

- `await` in the initialization of `for` statement
- `await` in the right-hand of `for-of` statement
- `await` in the right-hand of `for-in` statement
- `await` in the right-hand of `for-await-of` statement

and fixed the implementation to pass the tests.

**Pull request merged**